kernel/interrupt_controller/src/x86_64.rs

```diff
@@ -14,35 +15,38 @@ pub struct LocalInterruptControllerId(pub u32);
 #[derive(Debug, Copy, Clone)]
 pub struct Priority;
 
-/// Initializes the interrupt controller (not yet used on x86)
-pub fn init() -> Result<(), &'static str> { Ok(()) }
+/// Initializes the interrupt controller(s), including the Local APIC for the BSP
+/// (bootstrap processor) and the system-wide IOAPIC(s).
+pub fn init(kernel_mmi: &memory::MmiRef) -> Result<(), &'static str> {
+    apic::init();
+
+    // Use the MADT ACPI table to initialize more interrupt controller details.
+    {
+        let acpi_tables = acpi::get_acpi_tables().lock();
+        let madt = Madt::get(&acpi_tables)
+            .ok_or("The required MADT ACPI table wasn't found (signature 'APIC')")?;
+        madt.bsp_init(&mut kernel_mmi.lock().page_table)?;
```

Comment:
The API would be more consistent if this function obtained the page table from `memory` directly rather than getting it via call parameters. It would reduce the risk of deadlocks too? (here, not really)

Reply:
hmm, i disagree actually --- in general, it's better state management (and clearer expression of dependencies) to pass a function everything it needs in order for it to execute. Obviously that's not always possible, but at least in other initialization code so far we do try to always pass in things like MMI.

Reply:
Also marginally more efficient, haha. Nothing significant, but it doesn't hurt to avoid the attempt to fallibly access global kernel mmi state.